### PR TITLE
vpci: fix build with PCI_PASSTHROUGH/VIRTIO_PCI off

### PR DESCRIPTION
**Pull request by GrygiriiS**
Fix build with CONFIG_PCI_PASSTHROUGH/VIRTIO_PCI off.

Pretty brutal fixes. Tested.

### CODE REVIEW

**Comment by GrygiriiS** [comment]
@otyshchenko1 @lorc @Deedone

**Comment by lorc** [comment]
Could you please mark it as `[HACK]` at least? It will be helpful in the future, when we'll try to assemble proper solution from all these patches

File: xen/include/xen/vpci.h [hunk 1]
```diff
@@ -335,6 +335,8 @@ static inline bool __must_check vpci_process_pending(struct vcpu *v)
 #ifdef CONFIG_HAS_VPCI_GUEST_SUPPORT
 bool vpci_translate_virtual_device(const struct domain *d, pci_sbdf_t *sbdf);
 #else
+#include <asm/pci.h>
```
Comment (xakep-amatop) [inline xen/include/xen/vpci.h:338]: Wouldn't it be better to move the same line of code on line number 6 in this file above the condition `#ifdef CONFIG_HAS_VPCI`?

Comment (xakep-amatop) [reply xen/include/xen/vpci.h:338]: sorry, I didn't pay attention that we use xen/pci.h instead of asm/pci.h there

Comment (GrygiriiS) [reply xen/include/xen/vpci.h:338]: the problem is `#define is_hardware_pci_domain(d) (is_hardware_domain(d) && !hwdom_uses_vpci())`
few lines below. In particular hwdom_uses_vpci().

Potentially it just `false` if VPCI=n

**Pull request merged**